packages/stream_chat_flutter/lib/src/message_input.dart

```diff
@@ -472,7 +473,28 @@ class MessageInputState extends State<MessageInput> {
         child: child,
       );
     }
-    return child;
+    return AnchoredOverlay(
+      showOverlay: _commandsOverlay != null ||
+          _emojiOverlay != null ||
+          _mentionsOverlay != null,
```

Comment:
Can't we create 3 booleans for this instead?

Reply:
I suggested we create three different widget instead of using just one Overlay this way
What do you think about this?

For example the `EmojiOverlay` could have all the logic about picking emojis 

I think it falls also under the spike call we had yesterday

Reply:
Unlike the earlier impl, in this case we only use these booleans twice and only in the same code block, so LOC would not change either way, I'd suggest this being as it is.